# HG changeset patch
# User alta88@fixall.com
# Date 1685996295 -7200
# Parent  4d5e30af0b5feba731b23c3b2e90785848420806
Fix feed vandalization, Part I - folder/icon css
- lower opacity for folder name, not icon, when paused
- no opacity change for non feed folders
- increase visibility of error icon
- no !important (a poor css solution) for css "content"
- error, busy, new message indicators on the icon are css "content" (for screen readers);
  the icon url is not - it is "background-image"
- reenable error/busy "content" icons to overlap with new messages indicator
- no "fill" to override a site favicon
- for folders with multiple subscriptions, use a favicon (of first sub url found)

diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -3609,29 +3609,36 @@ class FolderTreeRow extends HTMLLIElemen
    * Sets folder properties based on the folder for the row.
    *
    * @param {nsIMsgFolder} folder
    */
   setFolderPropertiesFromFolder(folder) {
     if (folder.server.type != "rss") {
       return;
     }
-    let urls = !folder.isServer ? FeedUtils.getFeedUrlsInFolder(folder) : null;
-    if (urls?.length == 1) {
-      let url = urls[0];
-      this.icon.style = `content: url("page-icon:${url}"); background-image: none;`;
+    let urls = FeedUtils.getFeedUrlsInFolder(folder);
+    if (urls) {
+      this.dataset["isFeedFolder"] = "true";
+    } else {
+      delete this.dataset["isFeedFolder"];
     }
     let props = FeedUtils.getFolderProperties(folder);
     for (let name of ["hasError", "isBusy", "isPaused"]) {
       if (props.includes(name)) {
         this.dataset[name] = "true";
       } else {
         delete this.dataset[name];
       }
     }
+    if (!urls || props.includes("hasError") || props.includes("isBusy")) {
+      this.icon.style = "";
+    } else {
+      let url = urls[0];
+      this.icon.style = `background-image: url("page-icon:${url}");`;
+    }
   }
 
   /**
    * Update this row's name label to match the new `prettyName` of the server.
    *
    * @param {string} name
    */
   setServerName(name) {
diff --git a/mail/themes/shared/mail/about3Pane.css b/mail/themes/shared/mail/about3Pane.css
--- a/mail/themes/shared/mail/about3Pane.css
+++ b/mail/themes/shared/mail/about3Pane.css
@@ -492,33 +492,42 @@ menuitem.server[data-server-type="rss"] 
   --icon-color: var(--folder-color-folder-filter);
 }
 
 #folderTree li[data-server-type="nntp"] ul .icon {
   background-image: var(--icon-newsletter);
   --icon-color: var(--folder-color-newsletter);
 }
 
-#folderTree li[data-is-paused="true"] > .container > .icon,
-#folderTree li[data-is-paused="true"] > .container > .name,
-#folderTree li[data-is-paused="true"] ul > li > .container > .icon,
-#folderTree li[data-is-paused="true"] ul > li > .container > .name {
+#folderTree li[data-is-feed-folder] > .container > .icon {
+  background-image: var(--folder-pane-folder-rss);
+  background-size: 16px 16px;
+}
+
+#folderTree li[data-is-feed-folder]:not([data-is-busy]):not([data-has-error]) > .container > .icon {
+  fill: currentColor;
+}
+
+#folderTree li[data-is-paused] > .container > .name,
+#folderTree li[data-server-type][data-is-paused] ul > li[data-is-feed-folder] > .container > .name {
   opacity: 0.6;
 }
 
-#folderTree li[data-is-busy="true"] > .container > .icon {
-  content: var(--icon-clock) !important;
-  background-image: none;
+#folderTree li[data-is-busy] > .container > .icon,
+#folderTree li[data-has-error] > .container > .icon {
+   background-image: none;
+ }
+
+#folderTree li[data-is-busy] > .container > .icon::before {
+  content: var(--icon-clock);
 }
 
-#folderTree li[data-has-error="true"] > .container > .icon {
-  content: var(--icon-warning) !important;
-  background-image: none;
-  fill: var(--color-amber-30);
-  stroke: var(--color-amber-60);
+#folderTree li[data-has-error] > .container > .icon::before {
+  content: var(--icon-warning);
+  fill: var(--color-red-60);
 }
 
 #folderTree li[data-tag-key] > .container > .icon {
   background-image: var(--icon-tag);
 }
 
 .name {
   flex: 1;
